python/pyiceberg/catalog/base.py

```diff
@@ -161,12 +161,12 @@ def drop_namespace(self, namespace: str | Identifier) -> None:
             namespace: Namespace identifier
 
         Raises:
-            NamespaceNotFoundError: If a namespace with the given name does not exist
+            NoSuchNamespaceError: If a namespace with the given name does not exist
             NamespaceNotEmptyError: If the namespace is not empty
         """
 
     @abstractmethod
-    def list_tables(self, namespace: str | Identifier | None = None) -> list[Identifier]:
+    def list_tables(self, namespace: str | Identifier) -> list[Identifier]:
```

Comment:
I think that having `None` here was correct. You can list tables in the root namespace.

Reply:
Technically I understand it is possible, but this isn't the case for Hive and also not for the REST catalog (namespace is required): https://github.com/apache/iceberg/blob/master/open-api/rest-catalog-open-api.yaml#L401

Therefore I decided to remove it for now. WDYT?

Reply:
This isn't a change I would mix in with the current PR, and I probably wouldn't bother making it anyway. The `Catalog` contract does should allow this, so I don't see a reason to remove it from the contract documentation just because we don't have an implementation in the Python code yet.